pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1090,12 +1090,29 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
-// GetAllocatableDevices returns information about all the devices known to the manager
 func (m *ManagerImpl) GetAllocatableDevices() ResourceDeviceInstances {
 	m.mutex.Lock()
-	resp := m.allDevices.Clone()
```

Comment:
If you don't mind losing some of the extra logging you introduced in this function, I think a better / more concise implementation would be to extend the `ResourceDeviceInstances` type to attach a `Filter()` function that takes a `map[string]sets.String` and returns a new  `ResourceDeviceInstances` with only the filtered set of devices in it.

That would leave this function mostly unchanged, other than to transform the single line above to:
```
resp := m.allDevices.Filter(m.healthyDevices)
```

Reply:
That's a nice idea. Done, and added one extra unit test.